<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Stephen Hardies Phage Protein Families</title>
  </head>
  <body>
    <main>
        <h1>Stephen Hardies Phage Protein Families</h1>  
        <p>The purpose of this site is to provide public access to protein family alignments, HMM models, and other supplementary information in support of papers published by Stephen C. Hardies.  Since I expand my families past the limits of the most commonly used alignment software, other investigators will have trouble confirming or extending my studies without these resources.  Some of these models may eventually appear in Pfam.  I mention Pfam specifically, because that is the most accessible route for an investigator to incorporate an alignment into a public database.  However, others of my alignments do not fit the Pfam specifications concerning specificity and overlap.  For example, a key alignment for my purposes is that of all large terminases in tailed phages.  In order to broaden the alignment to include all the different phage families, it will also give significant matching to other P-loop ATPases, and other ruvC paralogs.  In order to use the HMM as an alignment tool to make timetrees across the breadth of the tailed phage viriome, I need them all in a common alignment.  Pfam prefers to maintain multiple large terminase subfamilies.  For other phage proteins, I typically try to expand their trees as deeply as the available techniques will allow.  Hence, although most phage proteins are not amenable to a fully global alignment, this problem of an expanded alignment versus the typical protein family alignment arises repeatedly in my work. Further I typically expand each family ahead of each use, rather than using a static alignment.</p>

<p>See <a href="GT1pap1.generic.technical.info.html">General technical information</a> for details about the methodology employed.



         <p><a href="GT1pap1.html">Families used in Stephen C. Hardies, Byung C. Cho, Gwang I Jang, Zhiquing Wang, Chung Y. Hwang. Identification of structural and morphogenesis genes of
      <i>Sulfitobacter</i> phage phiGT1 and placement within the evolutionary history of the podoviruses</a>.  Viruses 2023, 15(7), 1475; <a href="https://doi.org/10.3390/v15071475">https://doi.org/10.3390/v15071475</a>.
<br>
<p>Files in support of Chung Y. Hwang Byung C. Cho, Jin K. Kang, Jihye Park, Stephen c. Hardies.  Genomic Analysis of Two Cold-Active <i>Pseudoalteromonas</i> Phages Isloated from the Contenental Shelf in the Arctic Ocean.   To be submitted prsently.
<ul>
<li><a href="ACA1.genbank">ACA1.genbank</a> (OR438238)
<li><a href="ACA2.genbank">ACA2.genbank</a> (OR438929)
<li><a href="proACA1-A.genbank">proACA1-A.genbank</a> (OR438930)
</ul>
    </main>
  </body>
</html>
